kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/spec/SpecExecutor.kt

```diff
@@ -62,7 +62,7 @@ class SpecExecutor(private val listener: TestEngineListener) {
    private suspend fun runTestsIfAtLeastOneActive(spec: Spec): Try<Map<TestCase, TestResult>> {
       log("runTestsIfAtLeastOneActive [$spec]")
       val roots = spec.materializeAndOrderRootTests()
-      val active = roots.any { it.testCase.isActive() }
+      val active = roots.any { it.testCase.isActive().active }
       return if (active) runTests(spec) else emptyMap<TestCase, TestResult>().success()
```

Comment:
I think this shouldn't return emptyMap, it should instead return the skipped test results 🤔 
Maybe do this based on project config?

Reply:
skipped tests are kind of tricky. JUnit does not allow you to consider them as "started", so the flow is

for a passed test => test started, test finished with success
for a failed test => test started, test finished with error
for a skipped test => test ignored

It's weird API imo, they should have done the latter as

test started, test finished with ignore flag

Which is how kotest does it.

I'm ok changing the internals of kotest to return all tests with their state, just make sure it doesn't have any unforeseen issues elsewhere (it shouldn't as this isn't exposed to users).

Reply:
I ended up adding a [specSkipped listener](https://github.com/kotest/kotest/pull/2173/files#diff-4fa1869c048f6988cbb8e8fd07d85f3b6ab0dabdcab67bba0f7ddeaf84f75881R68) and left the return result of this function alone.